File: app/src/main/java/ru/nikshlykov/englishwordsapp/GroupDao.java
package ru.nikshlykov.englishwordsapp;

import androidx.room.Dao;
import androidx.room.Query;

@Dao
public interface GroupDao {

    @Query("SELECT * FROM Groups")
    Group[] getAllGroups();

    @Query("SELECT * FROM Groups WHERE _id=:id")
    Group getGroupById(long id);
}

File: app/src/main/java/ru/nikshlykov/englishwordsapp/db/daos/GroupDao.kt
package ru.nikshlykov.englishwordsapp.db.daos

import androidx.room.Dao
import androidx.room.Query
import ru.nikshlykov.englishwordsapp.db.models.Group

@Dao
interface GroupDao {

  @Query("SELECT * FROM Groups")
  suspend fun getAllGroups(): List<Group>
}